plugins/follow_me/follow_me.h

```diff
@@ -0,0 +1,170 @@
+#pragma once
+
+#include <iostream>
+#include <cmath>
+#include <functional>
+
+namespace dronecore {
+
+class FollowMeImpl;
+
+/**
+ * @brief FollowMe plugin is used to make your drone to follow your Ground Control Station (Phone/Tab). <br>
+ *
+ * Application need to obtain current GCS position from underying platform,
+ * and feed to it FollowMe, which makes your drone to move.
+ * When GCS keeps moving to a new position, position updates are sent to the drone
+ * accordingly; this results in drone movement relative to that of GCS.
+ *
+ * Please refer https://docs.px4.io/en/flight_modes/follow_me.html for more information.
+ *
+ */
+class FollowMe
+{
+public:
+    /**
+     * @brief Constructor (internal use only).
+     * @param impl Private internal implementation.
+     */
+    explicit FollowMe(FollowMeImpl *impl);
+
+    /**
+     * @brief Destructor (internal use only).
+     */
+    ~FollowMe();
+
+    /**
+     * @brief Results of FollowMe operations
+     */
+    enum class Result {
+        SUCCESS = 0, /**< @brief %Request succeeded. */
+        NO_DEVICE, /**< @brief No device connected. */
+        CONNECTION_ERROR, /**< @brief %Connection error. */
+        BUSY, /**< @brief Vehicle busy. */
+        COMMAND_DENIED, /**< @brief Command denied. */
+        TIMEOUT, /**< @brief %Request timeout. */
+        NOT_ACTIVE, /**< @brief FollowMe is not activated yet */
+        UNKNOWN /**< @brief Unknown error. */
+    };
+
+    /**
+     * @brief FollowMe Configuration.
+     * Reference: https://docs.px4.io/en/advanced_config/parameter_reference.html#follow-target
+     */
+    struct Config {
+        /**
+         * @brief Side to follow target from.
+         */
+        enum class FollowDirection {
+            FRONT_RIGHT, /**< @brief Follow from front right. */
+            BEHIND, /**< @brief Follow from behind. */
+            FRONT, /**< @brief Follow from front. */
+            FRONT_LEFT, /**< @brief Follow from front left. */
+            NONE
+        };
+        // Default Configuration
+        static const float DEF_HEIGHT_M; /**< @brief Default height above home: 8.0 mts */
```

Comment:
For the default, can't you just make it the default initialization of the config struct?

Reply:
Definitely I can. It will be messy in the public header file if line length becomes too long. This enables auto-warpping of the source code which decreases readability.

Reply:
What about having the comments before the field?

Reply:
> What about having the comments before the field?

What has been done here with comments is consistent with every other plugin so perhaps leave as is?

Reply:
```
        /**< @brief Default height above home, in meters. */
        constexpr static const float DEFAULT_HEIGHT_M = 8.0;
        /**< @brief Default follow distance, in meters. */
        constexpr static const float DEFAULT_FOLLOW_DIST_M = 8.0;
        /**< @brief Default follow direction. */
        constexpr static const FollowDirection DEFAULT_FOLLOW_DIR = FollowDirection::BEHIND;
        /**< @brief Default responsiveness to target. */
        constexpr static const float DEFAULT_RESPONSIVENSS = 0.5f;

        /**< @brief Min follow height, in meters. */
        constexpr static const float MIN_HEIGHT_M = 8.0f;
        /**< @brief Min follow distance from, in meters. */
        constexpr static const float MIN_FOLLOW_DIST_M = 1.0f;
        /**< @brief Min responsiveness. */
        constexpr static const float MIN_RESPONSIVENESS = 0.f;
        /**< @brief Max responsiveness. */
        constexpr static const float MAX_RESPONSIVENESS = 1.0f;

        /**< @brief Min follow height above home, in meters. */
        float min_height_m = DEFAULT_HEIGHT_M;
        /**< @brief Horizontal distance b/w vehicle & target. */
        float follow_dist_m = DEFAULT_FOLLOW_DIST_M;
        /**< @brief Relative position of the following vehicle. */
        FollowDirection follow_dir = DEFAULT_FOLLOW_DIR;
        /**< @brief Responsiveness: Range (0.0 to 1.0)*/
        float responsiveness = DEFAULT_RESPONSIVENSS;
```

Reply:
It would be like this.

Reply:
```
        constexpr static const float DEFAULT_HEIGHT_M =
            8.0; /**< @brief Default height above home, in meters. */
        constexpr static const float DEFAULT_FOLLOW_DIST_M =
            8.0; /**< @brief Default follow distance, in meters. */
        constexpr static const FollowDirection DEFAULT_FOLLOW_DIR =
            FollowDirection::BEHIND; /**< @brief Default follow direction. */
        constexpr static const float DEFAULT_RESPONSIVENSS =
            0.5f; /**< @brief Default responsiveness to target. */

        constexpr static const float MIN_HEIGHT_M = 8.0f; /**< @brief Min follow height, in meters. */
        constexpr static const float MIN_FOLLOW_DIST_M =
            1.0f; /**< @brief Min follow distance from, in meters. */
        constexpr static const float MIN_RESPONSIVENESS = 0.f; /**< @brief Min responsiveness. */
        constexpr static const float MAX_RESPONSIVENESS = 1.0f; /**< @brief Max responsiveness. */

        float min_height_m = DEFAULT_HEIGHT_M; /**< @brief Min follow height above home, in meters. */
        float follow_dist_m =
            DEFAULT_FOLLOW_DIST_M; /**< @brief Horizontal distance b/w vehicle & target. */
        FollowDirection follow_dir =
            DEFAULT_FOLLOW_DIR; /**< @brief Relative position of the following vehicle. */
        float responsiveness = DEFAULT_RESPONSIVENSS; /**< @brief Responsiveness: Range (0.0 to 1.0)*/
```
This is after in-line initialization.